hub/util/path.py

```diff
@@ -0,0 +1,24 @@
+from hub.core.storage import LocalProvider
+import os
+
+
+def local_provider_from_path(path: str):
+    """Construct a LocalProvider if given a valid local path, or return None
+
+    Arguments:
+        path (str): Path that may or may not indicate a local directory
+
+    Returns:
+        If the given path could point to a directory, return the LocalProvider.
+        If the given path does not look like a local path, return None.
+
+    Raises:
+        ValueError: If the given path is a local path to a file
+    """
+    if path.startswith((".", "/", "~")):
```

Comment:
you also can get away with not putting this in constants.py if it's part of the providers base class.

Reply:
Just gonna leave it there. Seems like a pretty one-off thing, I don't foresee much reuse outside this function